src/components/App/DisplayPuzzleInstances/index.tsx

```diff
@@ -0,0 +1,132 @@
+
+type DisplayPuzzleInstancesProps = {
+  puzzlesList: PuzzleCustom[];
+};
+
+const DisplayPuzzleInstances = ({
+  puzzlesList
+}: DisplayPuzzleInstancesProps) => {
+  const [puzzleId, setPuzzleId] = useState<string>('');
+  const [allPuzzleInstances, setAllPuzzleInstances] = useState<PuzzleInstance[]>([]);
+  const [puzzleInstances, setPuzzleInstances] = useState<
+    PuzzleInstance[]
+  >([]);
+
+  const getPuzzleInstances = async () => {
+    setAllPuzzleInstances(await getAllPuzzleInstances());
+  };
+
+  useEffect(() => {
+    getPuzzleInstances();
+  }, [puzzlesList]);
+
+  useEffect(() => {
+    if (!allPuzzleInstances || !setPuzzleInstances) return;
+
+    if (!puzzleId) return setPuzzleInstances(allPuzzleInstances);
+
+    setPuzzleInstances(
+      allPuzzleInstances.filter(instance => instance.puzzleId.toString() === puzzleId)
+    );
+  }, [puzzleId, allPuzzleInstances, setPuzzleInstances]);
+
+  return (
+    <div className={styles.puzzleInstancesWrapper}>
+      {/* HEADER */}
+      <div className={styles.puzzleInstancesHeader}>
+        <div className={styles.puzzleInstancesHeaderTitle}>
+          <h3> See Locations </h3>
+        </div>
+        {/* Select puzzle */}
+        <div className={styles.puzzleInstancesHeaderSelectWrapper}>
+          {puzzlesList.length > 0 ? (
+            <select
+              className={styles.puzzleInstancesHeaderSelect}
+              value={puzzleId}
+              onChange={e => setPuzzleId(e.currentTarget.value)}
+            >
+              <option value="">
+                All puzzles
+              </option>
+              {puzzlesList.map((puzzle: PuzzleCustom, index: number) => (
+                <option value={puzzle.id} key={`puzzle${index}`}>
+                  {puzzle.name}
+                </option>
+              ))}
+            </select>
+          ) : (
+            <select
+              className={styles.puzzleInstancesHeaderSelect}
+              value="No puzzles available"
+            ></select>
+          )}
+        </div>
+      </div>
+
+      {/* BODY */}
+      <div className={styles.puzzleInstancesBody}>
+        {/* 3 cols table title */}
+        <div className={styles.puzzleInstancesBodyTitle}>
+          <h4> Address </h4>
+          <h4> QR Code </h4>
+          <h4> Actions </h4>
+        </div>
+
+        {/* 3 cols table content */}
+        <div className={styles.puzzleInstancesBodyContent}>
+          {puzzleInstances.length > 0 &&
+            puzzleInstances.map(
+              (puzzleInstance: PuzzleInstance, index: number) => (
+                <div
+                  className={styles.puzzleInstancesBodyContentRow}
+                  key={`puzzleInstance${index}`}
+                >
+                  <div>{puzzleInstance.address}</div>
+                  <div>
+                    <QRGenerator
+                      className={styles.puzzleInstancesBodyContentQR}
+                      text={JSON.stringify(puzzleInstance)}
```

Comment:
Oh wait isn't the QR supposed to be the link to the puzzle instance? We'd need a function to generate that link